server/webapp/WEB-INF/applicationContext-acegi-security.xml

```diff
@@ -260,6 +260,8 @@
                 /admin/materials/**=ROLE_SUPERVISOR,ROLE_GROUP_SUPERVISOR
                 /admin/package_repositories/**=ROLE_SUPERVISOR,ROLE_GROUP_SUPERVISOR
                 /admin/package_definitions/**=ROLE_SUPERVISOR,ROLE_GROUP_SUPERVISOR
+                /api/admin/pipelines=ROLE_SUPERVISOR,ROLE_GROUP_SUPERVISOR
+                /api/admin/pipelines/*=ROLE_SUPERVISOR,ROLE_GROUP_SUPERVISOR
```

Comment:
All the above methods are applicable to ROLE_SUPERVISOR and ROLE_GROUP_SUPERVISOR alone. ROLE_USER does not have any permissions as far as any config in Go is concerned. They may/may-not get view/operate permissions for pipeline-runs on dashboard, but never an access to the pipeline config itself. 

Having said that, the weird part of this config that comes into picture is for `POST .../pipelines` which ends up creating a group when it does not exist. Creation of a new group should be allowed for `ROLE_SUPERVISOR` only. I don't know if there is a decent way to configure that in `applicationContext-acegi-security.xml`. There is `PreAuthorize` tag which seem to do something that we would want in this case, but I haven't given it a try myself.